### DISCUSSION TITLE
Do serverless function logs include edge-cached requests?

### DISCUSSION THREAD

**Discussion by rayshan** [vercel/community#63] [Help] [answered]
Hi! Questions:

1. Do serverless function logs include edge-cached requests?
2. If so, is there a way to filter logs based on what's edge-cached and what isn't?

I'm using Logflare as the log drain, and I see edge caching working well in the response header.

**Comment by paulogdm** [comment]
Hello,

1. Negative. We don't log cache-hits or misses, only Serverless Executions.
2. Nope. See above.

**Reply by rayshan** [reply]
Wait is this right? [Docs](https://vercel.com/docs/api#integrations/log-drains/json-drains) say:

`"cacheId": <original request id when request is served from cache>,`

And distinct counts of `requestId` and `cacheId` are different:

<img width="757" alt="Screen Shot 2020-08-18 at 11 01 27 AM" src="https://user-images.githubusercontent.com/3760840/90548687-2d296480-e142-11ea-995c-96833bd931e0.png">

**Reply by paulogdm** [reply]
You are correct. It seems we do have logs for those requests. My apologies.

**Comment by chasers** [comment]
![9B6FCFC2-0567-49BD-9C1F-66252ED9E23D](https://user-images.githubusercontent.com/1019814/90465005-4d660e80-e0c3-11ea-8cef-f99688bcbe6b.jpeg)

**Comment by paulogdm** [comment, marked as answer]
Final answer:

* All relevant requests accepted in Vercel will be piped if you are using "Log Drains". Using "Log Drains" is a **must** in production. 
* For "Real-time Requests", we only record Serverless Functions executions.

If you have any further questions, please [create your own discussion](https://github.com/vercel/community/discussions/new).

**Comment by chasers** [comment]
Ah like in your log UI you only get fn executions, thanks for clarifying.  I had actually noticed that UI was missing some stuff a while ago but didn't realize it was all the static requests.

**Comment by chasers** [comment]
Like does this make sense?

![image](https://user-images.githubusercontent.com/1019814/94297716-cefa5900-ff19-11ea-9a41-d7047fbeda2d.png)

https://vercel.com/integrations/logflare
